### PR TITLE
Redirect to variants page with message when variant is not found

### LINKED ISSUE #2448: typeError: 'NoneType' object is not subscriptable when trying to load a variant
```
/envs/prod/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/utils.py", line 24, in decorated_function
    context = func(*args, **kwargs)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/blueprints/variant/views.py", line 71, in sv_variant
    data["observations"] = observations(store, loqusdb, data["case"], data["variant"])
TypeError: 'NoneType' object is not subscriptable

```

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #2448 

**How to test**:
1. On scout stage (or local instance connected to LoqusDB), go to a variant (SV, SNV, STR) and copy the browser URL.
2. Paste this url omitting some characters, or replace the variant ID with a string of choice
3. Note that the page crashed
4. Repeat after installing this branch and make sure that page redirects with error without crashing

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR, DN

### CODE REVIEW

File: scout/server/blueprints/variant/views.py [hunk 1]
```diff
@@ -47,11 +47,10 @@ def variant(institute_id, case_name, variant_id):
 
     data = variant_controller(store, institute_id, case_name, variant_id=variant_id)
     if data is None:
-        LOG.warning(
-            "An error occurred: variants view requesting data for variant {}".format(variant_id)
-        )
         flash("An error occurred while retrieving variant object", "danger")
-        return redirect(request.referrer)
+        return redirect(
+            url_for("variants.variants", institute_id=institute_id, case_name=case_name)
```
Comment (dnil) [inline scout/server/blueprints/variant/views.py:52]: 👍

File: scout/server/blueprints/variant/views.py [hunk 2]
```diff
@@ -66,6 +65,12 @@ def sv_variant(institute_id, case_name, variant_id):
     """Display a specific structural variant."""
     data = variant_controller(store, institute_id, case_name, variant_id, add_other=False)
 
+    if data is None:
+        flash("An error occurred while retrieving variant object", "danger")
```
Comment (dnil) [inline scout/server/blueprints/variant/views.py:69]: 👍

File: scout/server/blueprints/variant/views.py [hunk 3]
```diff
@@ -85,7 +90,11 @@ def str_variant(institute_id, case_name, variant_id):
         add_other=False,
         get_overlapping=False,
     )
-
+    if data is None:
+        flash("An error occurred while retrieving variant object", "danger")
+        return redirect(
+            url_for("variants.str_variants", institute_id=institute_id, case_name=case_name)
```
Comment (dnil) [inline scout/server/blueprints/variant/views.py:96]: 👍

**Review by dnil — APPROVED** [review]
Works like a charm! Mental reminder - those are partially very similar controller functions. But that is for another PR.

**Comment by dnil** [comment]
✅ Screenshot 2021-03-22 at 19 45 28

**Comment by northwestwitch** [comment]
Thanks for the review @dnil!

**Pull request merged**